app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java

```diff
@@ -79,4 +95,30 @@ private void navigateToFragmentWithAction(int actionId) {
     public void onDestroyView() {
         super.onDestroyView();
     }
+
+    private void fetchUserImageFromDB(String userId) {
+        FirebaseStorage storage = FirebaseStorage.getInstance();
+        StorageReference storageReference = storage.getReference();
+
+        StorageReference ref = storageReference.child("images/" + userId + "/" + "userImage");
+
+        final long ONE_MEGABYTE = 1024 * 1024;
+
+        ref.getBytes(ONE_MEGABYTE).addOnSuccessListener(new OnSuccessListener<byte[]>() {
+            @Override
+            public void onSuccess(byte[] bytes) {
+                Bitmap bm = BitmapFactory.decodeByteArray(bytes, 0, bytes.length);
+                _userImage = bm;
+                _userImageView.setImageBitmap(_userImage);
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception e) {
+                Log.d(TAG, "Failed to get image");
+                // If the image hasn't finished uploading to the db yet, try again
+                fetchUserImageFromDB(userId);
+
+            }
+        });
+    }
```

Comment:
Seeing this method repeated in a whole bunch of places here. If we can extract it into a common file, that would be great and save on a lot of repetitive code.